homeassistant/components/roborock/coordinator.py

```diff
@@ -435,12 +443,28 @@ async def refresh_coordinator_map(self) -> None:
                 tasks.append(self.update_map())
             # If either of these fail, we don't care, and we want to continue.
             await asyncio.gather(*tasks, return_exceptions=True)
+            if self.data.status.in_cleaning:
```

Comment:
Can we do this up front? Decide a different set of map flags to iterate over up front -- pick the strategy, then execute rather than changing plans mid execution.